src/ios/SSZipArchive.m

```diff
@@ -81,8 +81,8 @@ + (BOOL)unzipFileAtPath:(NSString *)path toDestination:(NSString *)destination o
 	if ([delegate respondsToSelector:@selector(zipArchiveWillUnzipArchiveAtPath:zipInfo:)]) {
 		[delegate zipArchiveWillUnzipArchiveAtPath:path zipInfo:globalInfo];
 	}
-	if ([delegate respondsToSelector:@selector(zipArchiveProgressEvent:total:)]) {
-		[delegate zipArchiveProgressEvent:(NSInteger)currentPosition total:(NSInteger)fileSize];
+	if ([delegate respondsToSelector:@selector(zipArchiveProgressEvent:total:archivePath:)]) {
```

Comment:
this is only the objc layer and not the basic lib. If we don't change this, we need to create individual Delegates for each zip task, which would be a bit overkill.